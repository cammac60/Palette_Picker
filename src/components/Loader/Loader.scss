@import '../../variables.scss';

.loader {
  background: linear-gradient(270deg, #000000, #19d4ac);
  background-size: 400% 400%;
  height: 100%;
  padding-top: calc(25% - 50px);
  width: 100%;
  -webkit-animation: gradient-animation 10s ease infinite;
  animation: gradient-animation 10s ease infinite;
}

@-webkit-keyframes gradient-animation {
    0%{background-position:0% 50%}
    50%{background-position:100% 50%}
    100%{background-position:0% 50%}
}
@keyframes gradient-animation {
    0%{background-position:0% 50%}
    50%{background-position:100% 50%}
    100%{background-position:0% 50%}
}

.sk-cube-grid {
  width: 100px;
  height: 100px;
  margin: 0 auto;
}

.sk-cube-grid .sk-cube {
  width: 33%;
  height: 33%;
  background-color: #FFFFFF;
  float: left;
  -webkit-animation: sk-cubeGridScaleDelay 1.3s infinite ease-in-out;
          animation: sk-cubeGridScaleDelay 1.3s infinite ease-in-out;
}
.sk-cube-grid .sk-cube1 {
  -webkit-animation-delay: 0.2s;
          animation-delay: 0.2s; }
.sk-cube-grid .sk-cube2 {
  -webkit-animation-delay: 0.3s;
          animation-delay: 0.3s; }
.sk-cube-grid .sk-cube3 {
  -webkit-animation-delay: 0.4s;
          animation-delay: 0.4s; }
.sk-cube-grid .sk-cube4 {
  -webkit-animation-delay: 0.1s;
          animation-delay: 0.1s; }
.sk-cube-grid .sk-cube5 {
  -webkit-animation-delay: 0.2s;
          animation-delay: 0.2s; }
.sk-cube-grid .sk-cube6 {
  -webkit-animation-delay: 0.3s;
          animation-delay: 0.3s; }
.sk-cube-grid .sk-cube7 {
  -webkit-animation-delay: 0s;
          animation-delay: 0s; }
.sk-cube-grid .sk-cube8 {
  -webkit-animation-delay: 0.1s;
          animation-delay: 0.1s; }
.sk-cube-grid .sk-cube9 {
  -webkit-animation-delay: 0.2s;
          animation-delay: 0.2s; }

@-webkit-keyframes sk-cubeGridScaleDelay {
  0%, 70%, 100% {
    -webkit-transform: scale3D(1, 1, 1);
            transform: scale3D(1, 1, 1);
  } 35% {
    -webkit-transform: scale3D(0, 0, 1);
            transform: scale3D(0, 0, 1);
  }
}

@keyframes sk-cubeGridScaleDelay {
  0%, 70%, 100% {
    -webkit-transform: scale3D(1, 1, 1);
            transform: scale3D(1, 1, 1);
  } 35% {
    -webkit-transform: scale3D(0, 0, 1);
            transform: scale3D(0, 0, 1);
  }
}
