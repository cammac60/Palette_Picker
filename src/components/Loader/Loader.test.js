import React from 'react';
import Loader from './Loader';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import { shallow } from 'enzyme';

describe('Loader', () => {
  let wrapper;

  it('Should match the snapshot', () => {
    wrapper = shallow(<Loader />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render without crashing', () => {
    const div = document.createElement("div");
    ReactDOM.render(<Loader />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

});
