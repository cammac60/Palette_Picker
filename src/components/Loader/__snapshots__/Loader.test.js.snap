// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Should match the snapshot 1`] = `
<div
  className="loader"
>
  <div
    className="sk-cube-grid"
  >
    <div
      className="sk-cube sk-cube1"
    />
    <div
      className="sk-cube sk-cube2"
    />
    <div
      className="sk-cube sk-cube3"
    />
    <div
      className="sk-cube sk-cube4"
    />
    <div
      className="sk-cube sk-cube5"
    />
    <div
      className="sk-cube sk-cube6"
    />
    <div
      className="sk-cube sk-cube7"
    />
    <div
      className="sk-cube sk-cube8"
    />
    <div
      className="sk-cube sk-cube9"
    />
  </div>
</div>
`;
